### ISSUE TITLE
Multiple columns in a column family support works on read but not on write/update

### ISSUE THREAD

**Issue by heatherandthewho** [hortonworks-spark/shc#44]
**Summary: The catalog definition to help with movement from dataframe to hbase does not appear to consistently support having multiple columns associated to one column family when loading/saving data**

**1.  Write mode error when more than one column comes from same column family**
In spark shell define the catalog:
def empcatalog = s"""{
        |"table":{"namespace":"default", "name":"emp"},
        |"rowkey":"key",
        |"columns":{
          |"empNumber":{"cf":"rowkey", "col":"key", "type":"string"},
          |"city":{"cf":"personal data", "col":"city", "type":"string"},
          |"empName":{"cf":"personal data", "col":"name", "type":"string"},
          |"jobDesignation":{"cf":"professional data", "col":"designation", "type":"string"},
          |"salary":{"cf":"professional data", "col":"salary", "type":"string"}
        |}
      |}""".stripMargin

Define Case class:
case class HBaseRecordEmp(
empNumber:String,
city:String,
empName:String,
jobDesignation:String,
salary:String)

create some dummy data with spark and try to write and it says column family already created:
val data = (4 to 10).map { i =>  {
      val name = s"""Bobby${"%03d".format(i)}"""  
      HBaseRecordEmp(i,
      s"MyCity",
      name,
      "worker",
      "5000")
  }
}

sc.parallelize(data).toDF.write.options(
  Map(HBaseTableCatalog.tableCatalog -> catalog, HBaseTableCatalog.newTable -> "5")).format("org.apache.spark.sql.execution.datasources.hbase").save()

**ERROR:**
java.lang.IllegalArgumentException: Family 'professional' already exists so cannot be added
    at org.apache.hadoop.hbase.HTableDescriptor.addFamily(HTableDescriptor.java:829)

*_2.  Go to HBase shell and create the table with 2 column families, each with two columns as above manually and add some data, this works  on READ *_
def withCatalog(cat: String): DataFrame = {sqlContext.read.options(Map(HBaseTableCatalog.tableCatalog->cat)).format("org.apache.spark.sql.execution.datasources.hbase").load()}

val df = withCatalog(empcatalog)
df.show

**3.  Now that the table exists in Hbase with the 2 column families as expected, add some dummy data in the spark shell, and attempt the write again.  This will work if you change the SaveMode to "append".

**It seems like the hbase connector should support multiple columns in one column family as expected and this behavior is inconsistent.**

**Comment by 144Lucky** [comment]
As the example,I defined a catalog 
```
  val cat = s"""{
                |"table":{"namespace":"default", "name":"shcExampleTable"},
                |"rowkey":"key",
                |"columns":{
                |"col0":{"cf":"rowkey", "col":"key", "type":"string"},
                |"col1":{"cf":"**cf1**", "col":"col1", "type":"boolean"},
                |"col2":{"cf":"**cf1**", "col":"col2", "type":"double"},
                |"col3":{"cf":"cf3", "col":"col3", "type":"float"},
                |"col4":{"cf":"cf4", "col":"col4", "type":"int"},
                |"col5":{"cf":"cf5", "col":"col5", "type":"bigint"},
                |"col6":{"cf":"cf6", "col":"col6", "type":"smallint"},
                |"col7":{"cf":"cf7", "col":"col7", "type":"string"},
                |"col8":{"cf":"cf8", "col":"col8", "type":"tinyint"}
                |}
                |}""".stripMargin
```
when running,An error has occurred
Exception in thread "main" java.lang.IllegalArgumentException: Family 'cf1' already exists so cannot be added

**Comment by heatherandthewho** [comment]
@144Lucky  Exactly.   Create that table in hbase manually and you will be able to perform a read on it no problem.

**Comment by weiqingy** [comment]
Looking into this. Thanks for reporting this issue.

**Comment by 144Lucky** [comment]
@heatherandthewho . as **3  for the summary about u suggested,the method of append()  is not supported.can u give a example? 

`    sc.parallelize(data1).toDF.write.options(
      Map(HBaseTableCatalog.tableCatalog -> cat1))
      .format("org.apache.spark.sql.execution.datasources.hbase")
      .append()`

**Comment by heatherandthewho** [comment]
@144Lucky the append goes here, its the mode...it worked for me
 
def saveToHBase(sql: SQLContext, dataFrame: DataFrame, catalog:String) = {
     dataFrame.write.options(Map(HBaseTableCatalog.tableCatalog -> catalog, HBaseTableCatalog.newTable -> "5")).mode("append").format("org.apache.spark.sql.execution.datasources.hbase").save()
    }

**Comment by heatherandthewho** [comment]
As a future testing datapoint, since this is to eventually be pulled into the hbase-spark connector.... i also tried this with Cloudera CDH 5.8.0 .  I had to recompile the spark connector with hortonworks cdh5.8.0 libs, the "read" worked fine from existing tables.  On the save I was getting an error regarding getting a connection--almost like it wasn't reading the hbase-site.xml, though it seemed to pick it up on the read.

**Comment by weiqingy** [comment]
Fixed by [PR#77](https://github.com/hortonworks-spark/shc/pull/77), so close this. Thanks.
